examples/hello_vector/cpp/build.sh

```diff
@@ -1,34 +1,66 @@
 #!/bin/bash
 
+while [ $# -gt 0 ]; do
+    # echo arg $1
+    case "$1" in
+        TC=*)   # alternative to TOOLCHAIN env var
```

Comment:
To accommodate both worlds (power user & not), forcing verbosity on the command-line with TC variable does not satisfy the power-user.  Instead, if no TC variable is present on the command-line, then a check should be made for the $TOOLCHAIN evn.var.  If present, then we can infer something from the user's environment, otherwise we carry on as written.